### ISSUE TITLE
Application verification failed

### ISSUE THREAD

**Issue by LeapingSwan** [myflashlab/Firebase-ANE#258]
Adobe Air 30

Hi,

I am using ANELAB to try and install Firebase core into an existing project.  The project builds fine before including Firebase Core but as soon as I do include it the application will not install on my device and I get the message "Application Verification Failed".  I have tried cleaning everything out and rebuilding but get the same message.

I have copied below my xml file.

Any suggestions welcome.

Thanks

Stephen

XML File:
```xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<application xmlns="http://ns.adobe.com/air/application/30.0">
    <id>mypackagename</id>
    <versionNumber>1.0</versionNumber>
    <filename>ProductName</filename>
    <description></description>
    <name>ProductName</name>
    <copyright></copyright>
    <initialWindow>
        <content>ProductName.swf</content>
        <systemChrome>standard</systemChrome>
        <transparent>false</transparent>
        <visible>true</visible>
        <fullScreen>true</fullScreen>
        <autoOrients>true</autoOrients>
        <aspectRatio>portrait</aspectRatio>
        <renderMode>direct</renderMode>
    </initialWindow>
    <customUpdateUI>false</customUpdateUI>
    <allowBrowserInvocation>false</allowBrowserInvocation>
    <icon>
        <image29x29>icons/icon29.png</image29x29>
        <image57x57>icons/icon57.png</image57x57>
        <image114x114>icons/icon114.png</image114x114>
        <image512x512>icons/icon512.png</image512x512>
        <image48x48>icons/icon48.png</image48x48>
        <image72x72>icons/icon72.png</image72x72>
        <image50x50>icons/icon50.png</image50x50>
        <image58x58>icons/icon58.png</image58x58>
        <image100x100>icons/icon100.png</image100x100>
        <image144x144>icons/icon144.png</image144x144>
        <image1024x1024>icons/icon1024.png</image1024x1024>
        <image40x40>icons/icon40.png</image40x40>
        <image76x76>icons/icon76.png</image76x76>
        <image80x80>icons/icon80.png</image80x80>
        <image120x120>icons/icon120.png</image120x120>
        <image152x152>icons/icon152.png</image152x152>
        <image180x180>icons/icon180.png</image180x180>
        <image60x60>icons/icon60.png</image60x60>
        <image75x75>icons/icon75.png</image75x75>
        <image87x87>icons/icon87.png</image87x87>
        <image167x167>icons/icon167.png</image167x167>
    </icon>
    <supportedLanguages>en</supportedLanguages>
    <versionLabel></versionLabel>
    <android>
        <manifestAdditions><![CDATA[<manifest android:installLocation="auto" >
  <application android:name="android.support.multidex.MultiDexApplication">
    <activity>
      <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <data android:scheme="leapfrog.digital.bizbox"/>
      </intent-filter>
    </activity>
    <receiver android:name="com.google.android.gms.measurement.AppMeasurementReceiver" android:enabled="true" android:exported="false"/>
    <service android:name="com.google.android.gms.measurement.AppMeasurementService" android:enabled="true" android:exported="false"/>
    <service android:name="com.google.android.gms.measurement.AppMeasurementJobService" android:enabled="true" android:exported="false" android:permission="android.permission.BIND_JOB_SERVICE"/>
    <receiver android:name="com.google.android.gms.measurement.AppMeasurementInstallReferrerReceiver" android:enabled="true" android:exported="true" android:permission="android.permission.INSTALL_PACKAGES">
      <intent-filter>
        <action android:name="com.android.vending.INSTALL_REFERRER"/>
      </intent-filter>
    </receiver>
    <provider android:name="com.google.firebase.provider.FirebaseInitProvider" android:authorities="air.leapfrog.digital.bizbox.firebaseinitprovider" android:exported="false" android:initOrder="100"/>
    <service android:name="com.google.firebase.components.ComponentDiscoveryService">
      <meta-data android:name="com.google.firebase.components:com.google.firebase.iid.Registrar" android:value="com.google.firebase.components.ComponentRegistrar"/>
      <meta-data android:name="com.google.firebase.components:com.google.firebase.analytics.connector.internal.AnalyticsConnectorRegistrar" android:value="com.google.firebase.components.ComponentRegistrar"/>
    </service>
    <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version"/>
    <receiver android:name="com.google.firebase.iid.FirebaseInstanceIdReceiver" android:exported="true" android:permission="com.google.android.c2dm.permission.SEND">
      <intent-filter>
        <action android:name="com.google.android.c2dm.intent.RECEIVE"/>
        <action android:name="com.google.android.c2dm.intent.REGISTRATION"/>
        <category android:name="air.mypackagename"/>
      </intent-filter>
    </receiver>
    <service android:name="com.google.firebase.iid.FirebaseInstanceIdService" android:exported="true">
      <intent-filter android:priority="-500">
        <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
      </intent-filter>
    </service>
    <activity android:name="com.google.android.gms.common.api.GoogleApiActivity" android:theme="@android:style/Theme.Translucent.NoTitleBar" android:exported="false"/>
    <meta-data android:name="com.myflashlabs.aneLab.id" android:value="zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"/>
  </application>
  <uses-sdk android:minSdkVersion="15" android:targetSdkVersion="26"/>
  <uses-permission android:name="android.permission.INTERNET"/>
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
  <uses-permission android:name="android.permission.WAKE_LOCK"/>
  <uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE"/>
  <uses-permission android:name="com.google.android.c2dm.permission.RECEIVE"/>
  <uses-permission android:name="air.leapfrog.digital.bizbox.permission.C2D_MESSAGE"/>
</manifest>]]></manifestAdditions>
    </android>
    <iPhone>
        <requestedDisplayResolution>high</requestedDisplayResolution>
        <InfoAdditions><![CDATA[
  <key>MinimumOSVersion</key>
  <string>8.0</string>
  <key>com.myflashlabs.aneLab.id</key>
  <string>zzzzzzzzzzzzzzzzzzzzzzzzzzzzzz</string>
]]></InfoAdditions>
        <Entitlements><![CDATA[
  <key>get-task-allow</key>
  <true/>
  <key>application-identifier</key>
  <string>BBBBBBBBBB.mypackagename</string>
  <key>keychain-access-groups</key>
  <array>
    <string>BBBBBBBBBB.*</string>
  </array>
  <key>aps-environment</key>
  <string>development</string>
  <key>com.apple.developer.team-identifier</key>
  <string>BBBBBBBBBB</string>
]]></Entitlements>
    </iPhone>
</application>
```

**Comment by LeapingSwan** [comment]
By making the xml read only I have managed to isolate the offending element which is

<key>aps-environment</key>
  <string>development</string>

If I delete this the application does build, although I have yet to find if it has other consequences.  As I understand it this relate to push notifications which I am not using in this app, but might in the future so it would be good to get this cleared up.  I believe these lines are being created by ANELAB as I don't think I have any settings saying I need them but I can't be sure.

It also means as I add new ANEs I will have to keep removing these lines and resetting the xml to build only.

Thanks

Stephen

**Comment by LeapingSwan** [comment]
And then subsequently, installing DynamicLinks ANE (which I need as I am trying to use Firebase Auth to allow sign in) I had to exclude the following from the xml

 <key>com.apple.developer.associated-domains</key>
  <array>
    <string>applinks:XXXXX.app.goo.gl</string>
  </array>

or else again Application Verification failed.

The XXXXX is replacing the link I got from Firebase for deep links.  I think I picked the right code but I'm not sure.  Without this being in here I am not convinced that the ANE will work properly.

Now that I can at least build with the ANE's in I can try and implement them all.

Stephen
